src/chainstate/stacks/index/storage.rs

```diff
@@ -1395,13 +1057,17 @@ impl TrieFileStorage {
     }
 
     pub fn num_blocks(&self) -> usize {
-        self.block_map.len()
+        let result = if self.last_extended.is_some() {
+            1
+        } else {
+            0
+        };
+        result + (trie_sql::count_blocks(&self.db)
+                  .expect("Corruption: SQL Error on a non-fallible query.") as usize)
     }
     
     pub fn chain_tips(&self) -> Vec<BlockHeaderHash> {
-        let mut r = Vec::with_capacity(self.chain_tips.len());
-        r.extend(self.chain_tips.iter());
-        r
+        trie_sql::get_chain_tips(&self.db)
```

Comment:
I was under the impression that we didn't need to know or care about `chain_tips()` anymore, and that it was just an artifact of the old fork-table representation?  If so, wanna just remove this altogether?

Reply:
Yep, I'm all for removing it -- I'll poke around to see if anything's still calling it for some reason. I think the early integration of VM with the marf used it at some point, but I don't think it does anymore.

Reply:
Removed in 7116618